boost filesystem not found

### Report

/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
collect2: error: ld returned 1 exit status
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
make[2]: *** [src/topp/CMakeFiles/ClusterMassTracesByPrecursor.dir/build.make:108: bin/ClusterMassTracesByPrecursor] Error 1
make[1]: *** [CMakeFiles/Makefile2:4431: src/topp/CMakeFiles/ClusterMassTracesByPrecursor.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/AccurateMassSearch.dir/build.make:108: bin/AccurateMassSearch] Error 1
make[1]: *** [CMakeFiles/Makefile2:4233: src/topp/CMakeFiles/AccurateMassSearch.dir/all] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/FeatureFinderLFQ.dir/build.make:108: bin/FeatureFinderLFQ] Error 1
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
make[1]: *** [CMakeFiles/Makefile2:4365: src/topp/CMakeFiles/FeatureFinderLFQ.dir/all] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/ClusterMassTraces.dir/build.make:108: bin/ClusterMassTraces] Error 1
make[1]: *** [CMakeFiles/Makefile2:4398: src/topp/CMakeFiles/ClusterMassTraces.dir/all] Error 2
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/AssayGeneratorMetabo.dir/build.make:108: bin/AssayGeneratorMetabo] Error 1
make[1]: *** [CMakeFiles/Makefile2:4266: src/topp/CMakeFiles/AssayGeneratorMetabo.dir/all] Error 2
make[2]: *** [src/topp/CMakeFiles/BaselineFilter.dir/build.make:108: bin/BaselineFilter] Error 1
make[1]: *** [CMakeFiles/Makefile2:4332: src/topp/CMakeFiles/BaselineFilter.dir/all] Error 2
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/AssayGeneratorMetaboSirius.dir/build.make:109: bin/AssayGeneratorMetaboSirius] Error 1
make[1]: *** [CMakeFiles/Makefile2:4299: src/topp/CMakeFiles/AssayGeneratorMetaboSirius.dir/all] Error 2
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::detail::status(boost::filesystem::path const&, boost::system::error_code*)'
/usr/bin/ld: ../../lib/libOpenMS.so: undefined reference to `boost::filesystem::path::operator/=(boost::filesystem::path const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [src/topp/CMakeFiles/CometAdapter.dir/build.make:109: bin/CometAdapter] Error 1
make[1]: *** [CMakeFiles/Makefile2:4464: src/topp/CMakeFiles/CometAdapter.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

--- a/cmake/build_system_macros.cmake
+++ b/cmake/build_system_macros.cmake
@@ -43,13 +43,9 @@ macro(find_boost)
     "1.81.1" "1.81.0" "1.81"
     "1.80.1" "1.80.0" "1.80"
     "1.79.1" "1.79.0" "1.79"
-    "1.78.1" "1.78.0" "1.78"
-    "1.77.1" "1.77.0" "1.77"
-    "1.76.1" "1.76.0" "1.76"
-    "1.75.1" "1.75.0" "1.75"
-    "1.74.1" "1.74.0" "1.74")
+    "1.78.1" "1.78.0" "1.78")
 
-  find_package(Boost 1.74.0 COMPONENTS ${ARGN} REQUIRED CONFIG)
+  find_package(Boost 1.78.0 COMPONENTS ${ARGN} REQUIRED CONFIG)
 
 endmacro(find_boost)
 

--- a/src/openms/CMakeLists.txt
+++ b/src/openms/CMakeLists.txt
@@ -163,6 +163,7 @@ if (ENABLE_TDL)
 endif()
 
 # Use std::filesystem instead of boost::filesystem in boost::process (avoids linking boost_filesystem)
+# Requires Boost >= 1.78 (where BOOST_PROCESS_USE_STD_FS was introduced)
 target_compile_definitions(OpenMS PRIVATE BOOST_PROCESS_USE_STD_FS)
 
 if (WITH_OPENTIMS)

